### PR TITLE
PWN-8395 - bridge send logic changes

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8395

## Description of Work

PWN-8395 - bridge send logic changes

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt [hunk 1]
```diff
@@ -272,15 +277,54 @@ class BridgeSendPresenter(
...
+    private fun countTotal(amount: BigDecimal, sendFee: SendFee?): BigDecimal {
+        val bridgeFee = sendFee as? SendFee.Bridge
+        val fees = getFeeList(bridgeFee)
+        val fee = if (calculationMode.getCurrencyMode() is CurrencyMode.Fiat.Usd) {
+            fees.sumOf {
+                it.amountInUsd.toBigDecimalOrZero()
+            }
+        } else {
+            fees.sumOf {
+                it.amountInToken
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt:306]: one liner here would be ok
fees.sumOf { it.amountInToken } or fees.sumOf(Fee::amountInToken)

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt [hunk 2]
```diff
@@ -272,15 +277,54 @@ class BridgeSendPresenter(
...
+        }
+        setTotalValue(totalFormatted)
+    }
+
+    private fun countTotal(amount: BigDecimal, sendFee: SendFee?): BigDecimal {
+        val bridgeFee = sendFee as? SendFee.Bridge
+        val fees = getFeeList(bridgeFee)
+        val fee = if (calculationMode.getCurrencyMode() is CurrencyMode.Fiat.Usd) {
+            fees.sumOf {
+                it.amountInUsd.toBigDecimalOrZero()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt:302]: same one liner

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt [hunk 3]
```diff
@@ -314,14 +359,36 @@ class BridgeSendPresenter(
...
-        calculationMode.updateTokenAmount(token.total)
+        calculationMode.updateTokenAmount(token.total - feeTotalInToken)
         view?.updateInputValue(calculationMode.formatInputAmount, true)
         val message = resources.getString(R.string.send_using_max_amount, token.tokenSymbol)
         view?.showToast(TextContainer.Raw(message))
     }
 
+    private fun getFeeTotalInToken(): BigDecimal {
+        return getFeeList().sumOf {
+            it.amountInToken
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt:373]: one liner would be ok

**Review by gslevinkov — APPROVED** [review]

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt [hunk 4]
```diff
@@ -314,14 +359,36 @@ class BridgeSendPresenter(
...
+        }
+    }
+
+    private fun getFeeTotalInUsd(): BigDecimal {
+        return getFeeList().sumOf {
+            it.amountInUsd.toBigDecimalOrZero()
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt:379]: `it.amountInUsd
.toBigDecimalOrZero()`

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt [hunk 5]
```diff
@@ -314,14 +359,36 @@ class BridgeSendPresenter(
...
+        }
+    }
+
+    private fun getFeeList(bridgeSendFee: SendFee.Bridge? = null): List<BridgeFee> {
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt:383]: you can drop down it to class func

**Review by Davran331122 — APPROVED** [review]

**Pull request merged**